snykTask/src/task-args.ts

```diff
@@ -102,31 +111,45 @@ class TaskArgs {
     return this.projectName;
   }
 
+  // validate based on testTypeSeverityThreshold applicable thresholds
   public validate() {
-    if (this.failOnThreshold) {
-      if (this.isNotValidThreshold(this.failOnThreshold)) {
-        const errorMsg = `If set, failOnThreshold must be '${Severity.CRITICAL}' or '${Severity.HIGH}' or '${Severity.MEDIUM}' or '${Severity.LOW}' (case insensitive). If not set, the default is '${Severity.LOW}'.`;
-        throw new Error(errorMsg);
-      }
+    const taskTestType = this.testType || TestType.APPLICATION;
+    const validTestTypes: any = Object.values(TestType);
+    const taskTestTypeThreshold =
+      testTypeSeverityThreshold.get(taskTestType) ??
+      testTypeSeverityThreshold.get(TestType.APPLICATION);
+
+    if (!validTestTypes.includes(taskTestType)) {
+      const warningMsgTestType = `Invalid testType specified. Defaulted to 'app'`;
+      console.log(warningMsgTestType);
     }
 
-    if (this.severityThreshold) {
-      if (this.isNotValidThreshold(this.severityThreshold)) {
-        const errorMsg = `If set, severityThreshold must be '${Severity.CRITICAL}' or '${Severity.HIGH}' or '${Severity.MEDIUM}' or '${Severity.LOW}' (case insensitive). If not set, the default is '${Severity.LOW}'.`;
+    if (this.failOnThreshold) {
+      if (
+        !taskTestTypeThreshold?.includes(this.failOnThreshold.toLowerCase())
+      ) {
+        const errorMsg = `If set, failOnThreshold must be one from [${taskTestTypeThreshold}] (case insensitive). If not set, the default is '${Severity.LOW}'.`;
         throw new Error(errorMsg);
       }
     }
```

Comment:
I'm not really sure what we are checking here with `this.failOnThreshold`. As far as I see at its definition:

```ts
class TaskArgs {
  testType: string | undefined = '';
...
  failOnThreshold: string = Severity.LOW;
```

it's initialised with `Severity.LOW` and is not an optional (`?`) so it will always be truthy. 

I think we could remove it so we remove one level of complexity

```ts
if (!taskTestTypeThreshold?.includes(this.failOnThreshold.toLowerCase())) {
  const errorMsg = `If set, failOnThreshold must be one from [${taskTestTypeThreshold}] (case insensitive). If not set, the default is '${Severity.LOW}'.`;
  throw new Error(errorMsg);
}
```

Reply:
This `failOnThreshold` check is required as it may be explicitly set by user at Task input. And this value needs to be among the ones applicable based on its `testType`.  And if an invalid `testType` was explicitly set, then it would be  validated based on an `app` testType without failing.